### PR TITLE
Add get action rule by action plan id api

### PR DESCRIPTION
**Pull request by tej99**
- Added get action rules by action plan id endpoint
- Added uuid column to action rule table for put calls (put api PR coming soon)

https://trello.com/c/j7zJo9GC/70-tbl123-scheduling-collection-exercise-event-dates-and-action-service-actionplan-job-trigger-redesign-xxl

### CODE REVIEW

**Comment by tej99** [comment]
[TBL123: Scheduling Collection Exercise event dates and Action Service actionplan/job trigger redesign (XXL)](https://trello.com/c/j7zJo9GC/70-tbl123-scheduling-collection-exercise-event-dates-and-action-service-actionplan-job-trigger-redesign-xxl)

File: src/main/java/uk/gov/ons/ctp/response/action/service/impl/ActionTypeServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+public class ActionTypeServiceImpl implements ActionTypeService {
+
+
+  @Autowired
+  private ActionTypeRepository actionTypeRepo;
+
+  @CoverageIgnore
+  @Override
+  public ActionType findActionType(final Integer actionTypeKey) {
+    log.debug("Entering findActionRules");
```
Comment (rjweeks) [inline src/main/java/uk/gov/ons/ctp/response/action/service/impl/ActionTypeServiceImpl.java]: I think we should delete this log statement.

File: src/main/java/uk/gov/ons/ctp/response/action/domain/repository/ActionRuleRepository.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+   * @param actionPlanId This is the case id
+   * @return List<ActionRule> This returns all action rules for the specified action plan id.
+   */
+  @Query(value = "select ar.actionrulepk, ar.actionplanfk, ar.actiontypefk, ar.name, ar.description, ar.daysoffset,"
+          + " ar.priority, ar.id "
+          + "from action.actionrule ar "
+          + "inner join action.actionplan ap on ar.actionplanfk = ap.actionplanpk "
+          + "where ap.id = :p_actionPlanId ;", nativeQuery = true)
+  List<ActionRule> findByActionPlanId(@Param("p_actionPlanId") UUID actionPlanId);
+
```
Comment (madeye-matt) [inline src/main/java/uk/gov/ons/ctp/response/action/domain/repository/ActionRuleRepository.java]: This is unnecessary complexity - I am pretty sure if you just add a method called findByActionPlanFK to this interface, it should do exactly the same job (JpaRepository automatically work out what is required from the method name so you don't have to write the code - there are some good examples of this in CaseRepository)

File: src/main/java/uk/gov/ons/ctp/response/action/endpoint/ActionRuleEndpoint.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+   * To build a list of ActionRuleDTOs from ActionRules populating the actionTypeName
+   *
+   * @param actionRules a list of ActionRules
+   * @return a list of ActionRuleDTOs
+   */
+  private List<ActionRuleDTO> buildActionRulesDTOs(final List<ActionRule> actionRules) {
+    final List<ActionRuleDTO> actionRulesDTOs = mapperFacade.mapAsList(actionRules, ActionRuleDTO.class);
+
+    int index = 0;
+    for (final ActionRule actionRule : actionRules) {
```
Comment (madeye-matt) [inline src/main/java/uk/gov/ons/ctp/response/action/endpoint/ActionRuleEndpoint.java]: You could build this into the mapper but ActionRule -> ActionRuleDTO probably isn't going to happen anywhere else so it should be ok

File: src/main/resources/database/changes/release-10.49.0/add_action_rule_uuid_column.sql [hunk 4]
```diff
@@ -0,0 +1,8 @@
+-- create postgres extension to allow generation of v4 UUID
+CREATE EXTENSION IF NOT EXISTS "pgcrypto";
+
+ALTER TABLE action.actionrule
+ADD COLUMN id uuid;
+
+UPDATE action.actionrule
+set id = gen_random_uuid()
```
Comment (madeye-matt) [inline src/main/resources/database/changes/release-10.49.0/add_action_rule_uuid_column.sql]: You probably want to add a UNIQUE & NOT NULL constraint on the id column after you've populated it

File: src/main/java/uk/gov/ons/ctp/response/action/domain/repository/ActionRuleRepository.java (hunk 2 above)
Comment (madeye-matt) [reply src/main/java/uk/gov/ons/ctp/response/action/domain/repository/ActionRuleRepository.java]: ok - didn't realise id wasn't primary key so you can ignore this one :-)

**Pull request merged**